src/pages/RepoPage/CoverageTab/CoverageTabNavigator.tsx

```diff
@@ -31,17 +32,25 @@ function CoverageTabNavigator() {
   const { provider, owner, repo } = useParams<URLParams>()
   const location = useLocation()
   const history = useHistory()
-
   const { coverage, flagsTab, componentsTab } = useNavLinks()
-  const urls = {
-    Overview: coverage.path({ provider, owner, repo }),
-    Flags: flagsTab.path({ provider, owner, repo }),
-    Components: componentsTab.path({ provider, owner, repo }),
-  }
+
+  const urls = useMemo(
+    () => ({
+      Overview: coverage.path({ provider, owner, repo }),
+      Flags: flagsTab.path({ provider, owner, repo }),
+      Components: componentsTab.path({ provider, owner, repo }),
+    }),
+    [coverage, flagsTab, componentsTab, owner, provider, repo]
+  )
+
+  const value = useMemo(
+    () => getSelection(location.pathname, urls),
+    [location.pathname, urls]
+  )
```

Comment:
Testing revealed a bug where redirecting off of flags/components back to overview would not update the navigator selection. Now the selection always matches the URL.